server/src/main/java/io/crate/auth/HostBasedAuthentication.java

```diff
@@ -141,6 +146,19 @@ SortedMap<String, HBAConf> convertHbaSettingsToHbaConf(Settings settings) {
         return Collections.unmodifiableSortedMap(hostBasedConf);
     }
 
+    @VisibleForTesting
+    Map<String, String> getJWTDefaults(Settings settings) {
+        // Returns empty string for absent values.
+        // This is needed for ImmutableMap construction
+        // and also helps to avoid NULL checks on token verification and instead fail on claim mismatch.
+        String iss = AuthSettings.AUTH_HOST_BASED_JWT_ISS_SETTING.get(settings);
+        String aud = AuthSettings.AUTH_HOST_BASED_JWT_AUD_SETTING.get(settings);
+        return Map.of(
+            ISS_KEY, iss,
+            AUD_KEY, aud
+        );
```

Comment:
Providing a map with empty strings in tests is cumbersome.
I think I should
-  use a regular Map which allows null values 
- swap setting extraction to actually get null, `settings.get(SETTING.getKey())` returns null for absent values
- adjust JWTAuthentication to deal with `issuer ==null`

Reply:
Why are you converting this to a map at all - couldn't we pass along the settings as is and leave the extraction up to the JWT auth method?

Reply:
Didn't want to propagate more info that we needed. But propagating settings is also fine, even removed a test because auth tests implicitly test resolve defaults logic.